lib/kino/input.ex

```diff
@@ -0,0 +1,228 @@
+defmodule Kino.Input do
+  @moduledoc """
+  Various input elements for entering data.
+
+  ## Examples
+
+  First, create an input and make sure it is rendered,
+  either by placing it at the end of a code cell or by
+  explicitly rendering it with `Kino.render/1`.
+
+      input = Kino.Input.text("Name")
+
+  Then read the value at any later point:
+
+      name = Kino.Input.read(input)
+  """
+
+  defstruct [:attrs]
+
+  @type t :: %__MODULE__{attrs: map()}
+
+  defp new(attrs) do
+    token = Kino.Bridge.generate_token()
+    id = {token, attrs} |> :erlang.phash2() |> Integer.to_string()
+    attrs = Map.put(attrs, :id, id)
+    %__MODULE__{attrs: attrs}
+  end
+
+  @doc """
+  Creates a new text input.
+
+  ## Options
+
+    * `:default` - the initial input value. Defaults to `""`
+
+    * `:multiline` - whether the input should be multiline
+      text box. Defaults to `false`
+  """
+  @spec text(String.t(), keyword()) :: t()
+  def text(label, opts \\ []) when is_binary(label) and is_list(opts) do
+    default = Keyword.get(opts, :default, "")
+    multiline = Keyword.get(opts, :multiline, false)
+    new(%{type: :text, label: label, default: default, multiline: multiline})
+  end
+
+  @doc """
+  Creates a new password input.
+
+  This is similar to text input, except the content is not
+  visible by default.
+
+  ## Options
+
+    * `:default` - the initial input value. Defaults to `""`
+  """
+  @spec password(String.t(), keyword()) :: t()
+  def password(label, opts \\ []) do
+    default = Keyword.get(opts, :default, "")
+    new(%{type: :password, label: label, default: default})
+  end
+
+  @doc """
+  Creates a new number input.
+
+  The input value is can be either a float or `nil`.
```

Comment:
Is it always a float? 🤔

Reply:
Yeah, it's a single number field so we always parse into float. Maybe we should have an option like `integer: true`?

Reply:
My understanding is actually the opposite, the number input is problematic on many browsers as not all of them support the decimal placeholder. We could try to read it as an integer and fallback to float only if the dot is there?

Reply:
My point was rather than if we get `"10"`, `"10.0"`, `"10.1"` we just normalize all to float, so the type is consistent. But we can definitely try parsing as integer then as float. I was thinking we could have an option that makes the data type clear, but then we can also leave any conversion up to the user?

Reply:
> we can also leave any conversion up to the user?

Yeah, that's my current thinking right now. They may also want to use Decimal, for example.